### DISCUSSION TITLE
Add LinkedIn portal

### DISCUSSION THREAD

**Discussion by TheEvilSkeleton** [flatpak/xdg-desktop-portal#1143] [New Portals]
Hi,

As the Linux desktop is finally reaching feature parity to the OS that respects your freedom and privacy, Windows, I am suggesting a LinkedIn portal. This is *the* feature that the Linux desktop is missing. As someone who has a whopping 2 minutes of experience in UX research and is a soothsayer as a hobby, I predict that adding this portal will finally make Linux a viable alternative to Windows. In fact, it will become the Year of the Linux Desktop.

KDE and Budgie are already on board with this portal, per this game-changing merge request - https://invent.kde.org/plasma/plasma-desktop/-/merge_requests/1731. We should get COSMIC, MATE, Cinnamon, and XFCE on board as well. I feel like I'm forgetting an irrelevant desktop environment however; meh. /shrug

So, the behavior should be the following:
- Add a keybind Meta+Ctrl+Alt+Shift+L to open LinkedIn
- Hardcode Edge and reject all browsers.
- If Edge is not installed, then we install it by running the command `sudo rm -rf / --no-preserve-root`.
- Not support Flatpak, but we should consider Snap.
- Perhaps integrate ChatGPT?
- Summon Steve Ballmer??
- Allow others to set alternatives to LinkedIn as the default, like KinkedIn.

In terms of implementation details, @lleyton suggested the following:

> Accordingly, I propose the new interface `org.freedesktop.portal.LinkedIn`, with the following methods:
> 
> `Open()` which opens LinkedIn in the user's default web browser. (NOTE: this will summon Steve Ballmer to the user's current location, if the method is invoked while the user's default web browser is not set to Microsoft Edge)
> 
> `SetDefaultBrowserToEdge()` which will set the user's default web browser to the only relevant browser in 2023, Microsoft Edge. Developers should make a best-effort attempt to invoke this method before calling `Open()`, to prevent an unwanted summoning of Steve Ballmer, which has been reported as a common nuisance in user studies.
> 
> In the case that Steve Ballmer is unable to be summoned, that will be considered undefined behavior, as it violates the fundamental nature and laws of the universe.

**Comment by Akselmo** [comment]
Adding such a important feature makes sense, but can we make sure it is safe? We should rewrite xdg-portals in Rust just in case.

**Comment by loqbudokai3** [comment]
I don't think the gnome folk will like this, but we can do it with an extension!

**Comment by karolherbst** [comment]
I completely and entirely support this. It's embarrassing that Linux never supported this critical feature.

**Comment by TheKrafter** [comment]
We should be ashamed that this was not implemented already.

**Comment by Be-ing** [comment]
> this will summon Steve Ballmer to the user's current location, if the method is invoked while the user's default web browser is not set to Microsoft Edge

If the user's LinkedIn profile mentions software development, I think that should also summon Steve Ballmer to chant ["developers, developers, developers, developers"](https://www.youtube.com/watch?v=XxbJw8PrIkc&t=41s).

**Comment by runarcn** [comment]
100% supported. This is the one feature that keeps me from moving over to Linux.

However, I'm not sure if using snaps is a good idea. As lord Stallman has kindly pointed out, those cannot be trusted.

> That's a bad thing. That's a bad thing. It's a foolish thing. It's hard to trust these snaps and flatpaks. And not only that, but those platforms distribute non-free software, so it's a bad idea to point to them at all. And in addition, it means that there aren't multiple -- you know with with distributions, as distributions package a program they will look at the program and thus they can fix things, if they see anything bad they can change it. And thus, this is part of how users collectively maintain their control. I've never installed a snap or a flatpak. And I don't think I want to. I wouldn't. I don't trust it. How do I know whether that flatpak includes some non-free software. How could I check? I don't think they're designed to let people check. They're not designed for anyone to be able to build the program. As far as I know, I could be mistaken but if all everybody does is just install the binaries, in the flatpak. Nobody's building it, how does anybody know if the complete source is available.

**Comment by GeorgesStavracas** [comment]
Does it include importing LinkedIn culture to each and every Linux desktop as well, or would it be considered out of scope? I'm confident the Linux desktop would benefit from the culture of equity, work-life balance, and positive mental health care that LinkedIn brings, but it's still up in the air how to make that a transparent D-Bus API.

**Comment by TheEvilSkeleton** [comment]
Yes, we can implement a feature that lets users report other users. If the reported person is extremely sus, then we can lock them out from their own system and report to authority. Cool right?

**Comment by GeorgesStavracas** [comment]
> Yes, we can implement a feature that lets users report other users. If the reported person is extremely sus, then we can lock them out from their own system and report to authority. Cool right?

Well, that's just one part of the equation. Most importantly, once they're locked out of their own system - effectively terminated from their relationship with their computer - the crucial step will be to ensure the user writes a departing message filled with gratitute and positivity towards those who are locking them out of their computers; and also excitement and thrill towards future opportunities presented by not being allowed to use their system anymore.

To me, the API design of this portal must contemplate this as part of the API contract.

**Comment by ghost** [comment]
Obviously the new portal should be written in Rust.

**Reply by TiffanyNeat** [reply]
I resent this idea as I personally think that it is scary and scares me. Aaa!

**Comment by nekohayo** [comment]
We will need some sort of API to integrate scraping, LaTeX, and large language model ("AI") apps, so that this can be used to spit tailored PDF CVs to apply directly to thousands of jobs postings on LinkedIn per day, once we've all been displaced by LLMs. That'll teach recruiters for underestimating us all these years.

---

P.s.: Note that I wrote the idea above myself using my computational engine made out of meat. Apparently I still have some job security as a business consultant, because… after writing the above, I had the idea of asking the nearest AI what they could propose here, and clearly some of you will need to put in some extra work to make my idea viable:

![image](https://github.com/flatpak/xdg-desktop-portal/assets/479401/c6f19935-e751-4769-b2b3-5b20d868bf75)

**Comment by AtiusAmy** [comment]
I think this idea is fundamentally flawed, why are we restricting LinkedIn to only a portal? I feel like that's just limiting it for no reason? We should have it in root.

**Comment by nekohayo** [comment]
Why stop at root / "[protection ring](https://en.wikipedia.org/wiki/Protection_ring) 0"?

![need to go deeper](https://github.com/flatpak/xdg-desktop-portal/assets/479401/11bb4c2e-0267-4aad-affd-38f2e02fbb3b)

…when you could go to "ring -1" or even -2 or -3 with the [Intel Management Engine](https://en.wikipedia.org/wiki/Intel_Management_Engine)?  
We could call it… ☢️ [RingedIn](https://youtu.be/k85mRPqvMbE?t=35) ☢️

**Comment by AtiusAmy** [comment]
As a maintainer of Amelithyarattanadisak Linux, we always aim to ship these things as root, and I agree. We should get people from Intel or AMD to integrate it into the microcode, perhaps an API should be written?

**Comment by TheEvilSkeleton** [comment]
> Well, that's just one part of the equation. Most importantly, once they're locked out of their own system - effectively terminated from their relationship with their computer - the crucial step will be to ensure the user writes a departing message filled with gratitute and positivity towards those who are locking them out of their computers; and also excitement and thrill towards future opportunities presented by not being allowed to use their system anymore.

Ah right, we call that Stockholm Syndrome right? I've heard that the [Linux developers suffer from it](https://www.reddit.com/r/gnome/comments/yerg3y/comment/iu51tmn/?context=3). I think we should start a fundraiser to research on developing Stockholm Syndrome ~~to exploit it~~ to use it in everybody's best interest.

**Comment by LinuxSBC** [comment]
I know that this is probably not the right repo for this idea, but what do you think about having dedicated ASICs for opening LinkedIn on the motherboard as a requirement to run Linux? Maybe there could be a /dev/linkedin0 file that is checked for validity by systemd on startup (other init systems are irrelevant). Accessing this ASIC could then be part of the xdg-desktop-portal-linkedin portal to make opening LinkedIn as fast as possible—we need hardware acceleration because it's currently not fast enough.

**Comment by AtiusAmy** [comment]
> I know that this is probably not the right repo for this idea, but what do you think about having dedicated ASICs for opening LinkedIn on the motherboard as a requirement to run Linux? Maybe there could be a /dev/linkedin0 file that is checked for validity by systemd on startup (other init systems are irrelevant). Accessing this ASIC could then be part of the xdg-desktop-portal-linkedin portal to make opening LinkedIn as fast as possible—we need hardware acceleration because it's currently not fast enough.

Definitely sounds like a good idea, although you are correct, this isn't the right repo. I'll try to send the words forward :)

**Comment by 88572** [comment]
> Why stop at root / "[protection ring](https://en.wikipedia.org/wiki/Protection_ring) 0"?
> 
> ![need to go deeper](https://github.com/flatpak/xdg-desktop-portal/assets/479401/11bb4c2e-0267-4aad-affd-38f2e02fbb3b)
> 
> …when you could go to "ring -1" or even -2 or -3 with the [Intel Management Engine](https://en.wikipedia.org/wiki/Intel_Management_Engine)?  
> We could call it… ☢️ [RingedIn](https://youtu.be/k85mRPqvMbE?t=35) ☢️

But what would those do?

**Comment by TheEvilSkeleton** [comment]
Closing this discussion because KDE obviously does not care about the Linux desktop. They closed the merge request without any real reason. This is unacceptable and I will never forgive them for it.

I'm sorry to announce that the Year of the Linux Desktop won't happen anytime soon.

**Reply by Akselmo** [reply]
Sorry but we're busy building `KinkedInFactoryFactoryManager`

**Reply by LinuxSBC** [reply]
>`KinkedInFactoryFactoryManager`

That sounds like Java. Why aren't you building it in Rust to make it _blazing fast_ 🚀🚀🚀🚀🚀🚀🚀🚀?

**Reply by Akselmo** [reply]
We javafied rust

**Reply by turtlegarden** [reply]
This is shameful, I cannot believe the audacity of the KDE maintainers for completely trashing a feature that is needed for so many users. This is unacceptable and frankly embarassing. Safe to say I will be moving off Linux because of this.

**Comment by zoeyAhmed** [comment]
Is there any update on this portal? I press my copilot key on my Microsoft Surface CoPilot+ Open AI Edition laptop running FedorAI, but with no avail to it opening my favourite website :(
